solutions/filedownloader.py

```
@@ -0,0 +1,220 @@
+
+    def initialize_download(self):
+        headers = self.request_and_receive_reponse_headers()
+        try:
+            if (int(headers['Content-Length']) == 0):
+                logging.warning('No data to download')
+        except:
+            logging.warning('Download size not available')
+            self.threads = 1
+        try:
+            if (headers['Accept-Ranges'] != 'bytes'):
+                self.threads = 1
+        except:
+            self.threads = 1
+
+    def get_request(self, https_method):
+        send_to_host = (
+            ' '.join(
+                (
+                    https_method,
+                    self.url_data.path,
+                    'HTTP/1.1'
+                )
+            )
+        )
+        send_to_host += EOL
+        headers = {}
+        headers['Host'] = self.url_data.host
+        headers['Connection'] = 'close'
+        # Add more headers when required
+        for key, value in headers.iteritems():
+            send_to_host += (
+                ''.join(
+                    (
+                        key,
+                        ': ',
+                        value,
+                        EOL
+                    )
+                )
+            )
+        return send_to_host
+
+    def request_and_receive_reponse_headers(self):
+        client_socket = socket.socket()
+        try:
+            client_socket.connect(
+                (
+                    self.url_data.host,
+                    int(self.url_data.port)
+                )
+            )
+        except:
+            logging.error('Client Socket connection could not be established')
+            raise
+        send_to_host = self.get_request('HEAD')
+        send_to_host += EOL
+        client_socket.send(send_to_host)
+        # Request is made only for headers
+        content_file = tempfile.NamedTemporaryFile()
+        received_headers = receive_data_from_host(client_socket, content_file)
```

How about you make the 2nd argument optional, so that you dont have to create an unnecessary temp file? That way, if you get anything beyond the headers, you can raise an exception.